design/secrets_provider_milestone1_design.md

```diff
@@ -0,0 +1,474 @@
+
+###### How does a Job answer the requirements?
+
+Running as a **Job** allows separation from the applications' deployment and serve multiple applications at once.
+When Secrets Provider finishes updating the K8s secrets, the **Job** completes successfully and the pod terminates.
+
+###### What benefits does this solution provide?
+
+1. **Clarity**
+   Allowing customers to select the deployment to get the desired value
+2. **Maintainability**
+   Using one image for all variations simplifies code maintenance 
+3. **Reusability**
+   Using one image for all variations allows reuse of existing code across all variations
+4. **Extensibility**
+   Supporting various flows using the same solution opens the door for more secrets sources and targets
+
+###### What drawbacks does this solution have?
+
+**Job** is not an existing application identity granularity.
+
+As a result, one can define the host representing the Secrets Provider in Conjur policy using only **namespace** and/or **service account** granularities.
+
+###### How the drawbacks can be handled?
+
+To handle the missing **Job** granularity there are 2 options:
+
+|      | Solution                                                     | Pros                                                         | Cons                                                         | Effort Estimation |
+| ---- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | ----------------- |
+| 1    | Keep using only **namespace** and/or **service account** granularities. | - Since Secrets Provider runs as a separate entity, we suggest  creating service account dedicated for Secrets Provider.<br />Then, **service account** granularity serves exact match and **Job** granularity is redundant<br />- No code changes | - Service account might be reused, which allows other apps to authenticate using Secrets Provider host | Free              |
+| 2    | Add support for **Job** granularity in Conjur                | - Allows specific identification of the Secrets Provider **Job**, which enhances security | - Costly; requires code changes, verify compatibility, tests, docs<br />- Redundant if **service account** serves only the Secrets Provider | 10 days           |
+
+*Decision*: Solution #1, use **namespace** and **service account** to define the Secrets Provider host in Conjur policy.
+Recommend to the customer to create a dedicated service account for Secrets Provider.
+
+#### Customer Experience
+
+1. Configure K8s Secrets with ***conjur-map\*** metadata defined 
+
+   - *Optional:* add label to K8s Secret
+
+2. Create Service Account for Secrets Provider 
+
+3. Create Role for the Service Accounts with **get**/**update** and also **list** privileges on K8s Secrets resources
```

Comment:
so the `list` privilege is new and comes from the labeling feature. You may be writing it later but I would give the label feature its own paragraph where you compare it with the current mechanism (`K8S_SECRETS` env var) including pros, cons and your decision. It is quite a big question that is not really coupled with this solution (serving multiple applications).

Reply:
don't forget backwards-compatibility.

Reply:
In the comparison table i would have columns from UX & security

Reply:
Yes `list` is new but again, it is optional to enhance UX. Adding under cons `- Demands "list" privileges on secrets`